Python/pythonrun.c

```
@@ -1253,183 +1253,221 @@ print_chained(struct exception_print_context* ctx, PyObject *value,
-                        if (err == 0) {
-                            err = write_indented_margin(ctx, f);
-                        }
-                        if (err == 0) {
-                            err = PyFile_WriteObject(line, f, Py_PRINT_RAW);
-                        }
-                        Py_DECREF(line);
-                    }
-                    else {
-                        err = -1;
-                    }
-                }
+            int err = PyFile_WriteObject(line, f, Py_PRINT_RAW);
+            Py_DECREF(line);
+            if (err < 0) {
+                return -1;
+            }
+        }
 
-                if (err == 0 && last_exc && ctx->need_close) {
-                    err = _Py_WriteIndent(EXC_INDENT(ctx), f);
-                    if (err == 0) {
-                        err = PyFile_WriteString(
-                            "+------------------------------------\n", f);
-                    }
-                    ctx->need_close = false;
-                }
-                ctx->exception_group_depth -= 1;
+        if (last_exc && ctx->need_close) {
+            if (_Py_WriteIndent(EXC_INDENT(ctx), f) < 0) {
+                return -1;
             }
+            if (PyFile_WriteString(
+                    "+------------------------------------\n", f) < 0) {
+                return -1;
+            }
+            ctx->need_close = false;
         }
-        if (ctx->exception_group_depth == 1) {
-            ctx->exception_group_depth -= 1;
+        ctx->exception_group_depth -= 1;
+    }
+
+    if (ctx->exception_group_depth == 1) {
+        ctx->exception_group_depth -= 1;
+    }
+    return 0;
+}
+
+static void
+print_exception_recursive(struct exception_print_context* ctx, PyObject *value)
+{
+    int err = 0;
+    if (ctx->seen != NULL) {
+        /* Exception chaining */
+        err = print_exception_cause_and_context(ctx, value);
+    }
+    if (err) {
+        /* don't do anything else */
+    }
```

Wouldn't it be clearer to just return if we don't need to do anything else? Actually, seeing that `err` is only set twice, I wonder if it could be worth it to just inline the error handling (`PyErr_Clear()`) and remove `err` 🤔 Nitpicking 😎

---

This will happen, once this function stops swallowing errors (that's a behaviour change rather than a refactor). Right now we have the PyErr_Clear() at the end.